specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/examples/DeploymentScripts_Create_using_existing_storage.json

```diff
@@ -0,0 +1,111 @@
+{
+  "parameters": {
+    "api-version": "2019-10-01-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "script-rg",
+    "scriptName": "MyDeploymentScript",
+    "deploymentScript": {
+      "kind": "AzurePowerShell",
+      "location": "westus",
+      "identity": {
+        "type": "UserAssigned",
+        "userAssignedIdentities": {
+          "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scriptRG/providers/Microsoft.ManagedIdentity/userAssignedIdentities/uai": {}
+        }
+      },
+      "properties": {
+        "azPowerShellVersion": "1.7.0",
+        "scriptContent": "Param([string]$Location,[string]$Name) $deploymentScriptOutputs['test'] = 'value' Get-AzResourceGroup -Location $Location -Name $Name",
+        "storageAccountSettings": {
+          "storageAccountName": "contosostorage",
+          "storageAccountKey": "contosostoragekey"
+        },
+        "arguments": "-Location 'westus' -Name \"*rg2\"",
+        "supportingScriptUris": [
+          "https://uri1.to.supporting.script",
+          "https://uri2.to.supporting.script"
+        ],
+        "retentionInterval": "PT7D",
+        "timeout": "PT1H",
+        "cleanupPreference": "Always"
+      }
+    }
+  },
+  "responses": {
+    "201": {
+      "body": {
+        "kind": "AzurePowerShell",
+        "location": "westus",
+        "identity": {
+          "type": "UserAssigned",
+          "userAssignedIdentities": {
+            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scriptRG/providers/Microsoft.ManagedIdentity/userAssignedIdentities/uai": {}
+          }
+        },
+        "systemData": {
+          "createdBy": "string",
+          "createdByType": "Application",
+          "createdAt": "2020-02-01T01:01:01.1075056Z",
+          "lastModifiedBy": "string",
+          "lastModifiedByType": "Application",
+          "lastModifiedAt": "2020-02-01T01:01:01.1075056Z"
+        },
+        "properties": {
```

Comment:
>properties [](start = 9, length = 10)

The response is missing the specified `storageAccountSettings`.

Reply:
Also, what are we going to return in the PUT response if you don't specify `storageAccountSettings`?

---
In reply to: [412423575](https://github.com/Azure/azure-rest-api-specs/pull/9157#discussion_r412423575) [](ancestors = 412423575)

Reply:
@majastrz added storageAccountSettings to the response and also updated the sample from container group name. 
If storageAccountSettings are not provided, null will be returned as the value of storageAccountSettings.

Question, in a response that contains storageAccountSettings -> storageAccountName, storageAccountKey is set as an empty string because it is marked as a required property, does this results in the property being returned with an empty string value or it will be ignored from the response?

Reply:
Properties that are secrets generally shouldn't be returned in the response or returned with explicit null (2nd case is rare). We shouldn't return empty string. The service can enforce that the storageAccountKey is present on PUT but it doesn't need to be returned in the response. I don't see any of these marked as required in the Swagger unless I'm missing something.

---
In reply to: [412492407](https://github.com/Azure/azure-rest-api-specs/pull/9157#discussion_r412492407) [](ancestors = 412492407)